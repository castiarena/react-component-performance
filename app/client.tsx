import React from 'react';
import { ThemeProvider } from 'styled-components';
import { hydrate } from 'react-dom';
import App from './App';
import theme from './theme';
import hotModuleReplacement from './infrastructure/Development/hotModuleReplacement';

const wrapper = document.getElementById('app-wrapper');

hydrate(
    <App />,
    wrapper,
);

hotModuleReplacement(module);
