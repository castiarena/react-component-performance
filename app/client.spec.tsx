describe('Client test suite', () => {
    it('should mount the app on the document with HRM', () => {
        const main = document.createElement('main');
        main.id = 'app-wrapper';
        document.body.appendChild(main);
        // eslint-disable-next-line global-require
        require('./client');
        expect(document.body.innerHTML).toMatchSnapshot();
    });
});
