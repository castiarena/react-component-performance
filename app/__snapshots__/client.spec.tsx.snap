// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client test suite should mount the app on the document with HRM 1`] = `"<main id=\\"app-wrapper\\"><div class=\\"sc-kEqYlL kXmtzB\\"><header class=\\"sc-pNWxx fhfGsX\\"><img src=\\"./assets/logo-realworldone.svg\\" alt=\\"Example\\"><a class=\\"sc-fujyUd sc-jrsJCI hfHqCh iLMECl\\" href=\\"/\\">Home</a><a class=\\"sc-fujyUd sc-jrsJCI hfHqCh iLMECl\\" href=\\"/comments\\">Comments</a><a class=\\"sc-fujyUd sc-jrsJCI hfHqCh iLMECl\\" href=\\"/performance\\">Performance</a></header><div><div class=\\"sc-bdnylx fCAKAP\\"><h1 class=\\"sc-hKFyIo dnWiOp\\">Real world one<strong>Frontend Developer Test</strong></h1><div class=\\"sc-bdnylx fSuCDU\\"><div class=\\"sc-iCoHVE dhneZx\\"><h2 class=\\"sc-eCApGN dbjYtj\\">Prerequisites</h2><p class=\\"sc-jSFkmK gqbJNj\\">Choose any JSON dummy data online generator. This will be your data source that will be used within your application (so called Test Data).</p><p class=\\"sc-jSFkmK gqbJNj\\">The Test Data size should be at least 500 records. e.g.<a href=\\"https://jsonplaceholder.typicode.com/comments\\" target=\\"_blank\\" class=\\"sc-gKAblj jrvysh\\">https://jsonplaceholder.typicode.com/comments</a></p></div><div class=\\"sc-iCoHVE kahQbH\\"><h2 class=\\"sc-eCApGN dbjYtj\\">Tasks</h2>Can see the full tasks on the PDF file:<p class=\\"sc-jSFkmK gqbJNj\\"><a href=\\"./assets/instructions.pdf\\" download=\\"\\" class=\\"sc-dlnjPT fTSCis\\">Download</a></p></div></div></div></div></div></main>"`;
